ui/hooks/snaps/useNavigation.js

```diff
@@ -0,0 +1,22 @@
+import { useHistory } from 'react-router-dom';
+import { parseMetaMaskUrl } from '@metamask/snaps-utils';
+import { getSnapRoute } from '../../helpers/utils/util';
+
+const useNavigation = () => {
+  const history = useHistory();
+  const navigate = (url) => {
+    let path;
+    const linkData = parseMetaMaskUrl(url);
+    if (linkData.snapId) {
+      path = getSnapRoute(linkData.snapId);
```

Comment:
As far as I can tell in the validation we don't validate that a Snap has a home page, so this could also link to the settings page? That seems unintentional.

Reply:
Yeah I was aware of that part, I kept as is because it's technically one route `/snaps/view...`. Someone might want to navigate to a snap's settings page as well.